src/dsp_tools/utils/excel_to_json/utils.py

```diff
@@ -0,0 +1,289 @@
+    # Remove the values of all cells that do not at least contain one character of any known language
+    cleaned_df = cleaned_df.applymap(
+        lambda x: str(x).strip() if pd.notna(x) and regex.search(r"[\w\p{L}]", str(x), flags=regex.U) else pd.NA
+    )
+    # drop all the rows that are entirely empty
+    cleaned_df.dropna(axis=0, how="all", inplace=True)
+    return cleaned_df
+
+
+def check_required_columns_raise_error(check_df: pd.DataFrame, required_columns: set[str]) -> None:
+    """
+    This function takes a pd.DataFrame and a set of required column names. It checks if all the columns from the set
+    are in the pd.DataFrame. Additional columns to the ones in the set are allowed. It raises an error if any columns
+    are missing.
+
+    Args:
+        check_df: pd.DataFrame that is checked
+        required_columns: set of column names
+
+    Raises:
+        UserError: if there are required columns missing
+    """
+    # This checks if the required columns are in the Dataframe. Other columns are also permitted.
+    if not required_columns.issubset(set(check_df.columns)):
+        raise UserError(
+            f"The following columns are missing in the excel: " f"{required_columns.difference(set(check_df.columns))}"
+        )
+
+
+def check_duplicate_raise_error(check_df: pd.DataFrame, duplicate_column: str) -> None:
+    """
+    This function checks if a specified column contains duplicate values. Empty cells (pd.NA) also count as duplicates.
+    If there are any duplicate values, it creates a string with the duplicates which are displayed in the error message.
+
+    Args:
+        check_df: pd.DataFrame that is checked for duplicates
+        duplicate_column: Name of the column that must not contain duplicates
+
+    Raises:
+        UserError: if there are duplicates in the column
+    """
+    # This checks if there are any duplicate values in a column,
+    # pd.NA values also count as duplicates if there are several empty cells.
+    if check_df[duplicate_column].duplicated().any():
+        # If it does, it creates a string with all the duplicate values and raises an error.
+        duplicate_values = ",".join(check_df[duplicate_column][check_df[duplicate_column].duplicated()].tolist())
+        raise UserError(
+            f"The column '{duplicate_column}' may not contain any duplicate values. "
+            f"The following values appeared multiple times '{duplicate_values}'."
+        )
+
+
+def check_required_values(check_df: pd.DataFrame, required_values_columns: list[str]) -> dict[str, pd.Series]:
+    """
+    This function takes a pd.Dataframe and a list of column names which may not contain empty cells. If there are any
+    empty cells in the column, it adds the column name and a boolean pd.Series to the dictionary. If there are no empty
+    cells, then it is not included in the dictionary. If no column has any empty cells, then it returns an empty
```

Comment:
A general feedback to your docstrings: I think that you should make a conceptual distinction between inline comments and docstrings:

- Inline comments explain implementation details. They are intended at other developers. 
- Docstring are 1 abstraction level higher: They are intended at users of the function. A docstring should enable the user to decide if he needs this function or not. It should explain to the outside world what tasks can be accomplished with this function.